docs/content/docs/dev/table/catalogs.md

```diff
@@ -738,3 +738,192 @@ Flink SQL> show tables;
 ```
 {{< /tab >}}
 {{< /tabs >}}
+
+## Catalog Store
+
+Catalog Store is used to store the configuration of catalogs. When using Catalog Store, the configurations
+of catalogs created in the session will be persisted in the corresponding external system of Catalog Store.
+Even if the session is reconstructed, previously created catalogs can still be retrieved from Catalog Store.
+
+### Configure Catalog Store
+Users can configure the Catalog Store in different ways, one is to use the Table API, and another is to use YAML configuration.
+
+{{< tabs "9bb8994c-7c07-4c49-f5b8-f4ae3548ead4" >}}
+{{< tab "Java/Scala" >}}
+```java
+// Register a catalog store using catalog store instance.
+
+// Initialize a catalog Store
+CatalogStore catalogStore = new FileCatalogStore("file://path/to/catalog/store/");
+
+// set up the catalog store
+final EnvironmentSettings settings =
+        EnvironmentSettings.newInstance().inBatchMode()
+        .withCatalogStore(catalogStore)
+        .build();
+
+final TableEnvironment tableEnv = TableEnvironment.create(settings);
+
+// Register a catalog store using configuration.
+        
+// Set up configuration
```

Comment:
Can we put them into different code blocks? It's a bit strange to put them together